packages/toolpad-app/src/runtime/ToolpadApp.tsx

```diff
@@ -1165,7 +1165,7 @@ function RenderedNodeContent({ node, childNodeGroups, Component }: RenderedNodeC
       NodeError={NodeError}
     >
       {isLayoutNode ? (
-        <Component {...wrappedProps} />
+        <Component {...wrappedProps} attributes={node.attributes} />
```

Comment:
Not sure about passing the attributes to every node here, they're supposed to have the component props only... 
Maybe let's just pass a different version of the `PageRootComponent` to `<RenderedNodeContent` depending on whether there is a container or not? One with the container layout and one with the layout without a container? That way we can just read from `page.attributes` inside the `RenderedPage` component.